docs/discussion/selection-api.md

```diff
@@ -0,0 +1,378 @@
+
+#### SelectRegion
+```
+def SelectRegion(region: Region) -> None
+```
+This method triggers a selection by region event based on the coordinates 
+
+  **Parameter**   |**Type**                                                                                            |**Description**
+  --------------- |---------------------------------------------------------------------------------------------------| -------------------------------------------------------
+  region          |[Region](https://astropy-regions.readthedocs.io/en/stable/api/regions.Region.html#regions.Region)   |The region corresponding to the sky coordinates that the user would like to select sources within
+  ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+#### SelectFootprint
+```
+def SelectFootprint(footprint: Footprint) -> None
+```
+This method triggers a selection event on a user provided footprint
+
+  **Parameter**   |**Type**    |**Description**
+  --------------- |----------- |---------------------------------------------
+  footprint       |Footprint   |The footprint the user would like to select
+
+  -------------------------------------------------------------------------
+
+# 3. aladin-lite interfaces
+
+## 3.1. Listeners
+
+Supported listeners that enable us to attach callback functions to
+events that occur within aladin-lite.
+
+  **Event Name**     |**Description**              |**Data**            |**Supported**
+  ------------------ |---------------------------- |------------------- |---------------
+  objectsSelected    |Fired when objects are selected by region, represents the individual objects selected |List of sources     |Existing
+  regionsSelected    |Fired when objects are selected by region, represents the region used to select the objects    |Shape of the region selected|New
+  objectClicked      |Fires when an object is clicked | Catalog object and xy coordinates of the click event | Existing
+  footprintClicked   |Fires when a footprint is clicked | Footprint object xy coordinates of the click event | Existing
+  -----------------------------------------------------------------------------------
+
+## 3.2. GET
+
+#### GetSelectedSources
+```
+/**
+ * @returns List[Sources]
+ */
+function GetSelectedSources()
+```
+Returns the list of currently selected source objects.
+
+#### GetSelectedRegion
+```
+/**
+ * @returns Region
+ */
+function GetSelectedRegion()
+```
+Returns the region definition of the selection event, if a region was
+used to select. 
+
+#### GetSelectedFootprints
+```
+/**
+ *@returns List[Footprint]
+ */
+function GetSelectedFootprints()
+```
+Returns the currently selected footprints
+
+## 3.3. SELECT
+
+#### SelectSources 
+
+##### Option 1: 
+```
+/**
+ * @param sources: List[Source]
+ * @param keys: List[str]
+ */
+function SelectSources(sources, keys)
```

Comment:
The details of how this would look are not clear to me, and they may not be very important yet, but an example of how it might look could be helpful.

Reply:
Here is what an [example implementation](https://github.com/pcuste1/ipyaladin/pull/2) might look like. I can add a link to the doc as well